app/models/trip.rb

```diff
@@ -0,0 +1,161 @@
+class Trip < ActiveRecord::Base
+
+  validates :duration,
+            :start_date,
+            :start_station_id,
+            :end_date,
+            :end_station_id,
+            :bike_id,
+            :subscription_type_id,
+            :user_zip_code,
+            :start_time,
+            :end_time,
+            presence: true
+
+  validates_uniqueness_of :bike_id, scope: [:start_date, :start_time]
+
+  belongs_to :start_station, class_name: 'Station', :foreign_key => :start_station_id
+  belongs_to :end_station, class_name: 'Station', :foreign_key => :end_station_id
+  belongs_to :subscription_type
+  belongs_to :weather_condition, :foreign_key => :start_date
+
+  ### Show Station methods ###
+  def self.number_rides_started_at_station(id)
+    where(start_station_id: id).count
+  end
+
+  def self.number_rides_ended_at_station(id)
+    where(end_station_id: id).count
+  end
+
+  def self.station_with_maximum_rides_as_start_location
+    station_counts = group(:start_station).count
+    station        = station_counts.max_by { |station, count| count }
+    "#{station.first.name} (#{station.last} rides)"
```

Comment:
this method has two lines repeated in the next method. let's break this down so we can reuse it rather than writing it twice 👍